src/lsp/sql_parser/sql_parser.ml

```diff
@@ -66,15 +65,10 @@ let parse text =
         (t, lstart, lend) :: tokens, Some lend
 
 
-      | pl ->
-        let t = Format.asprintf "%a" Cobol_preproc.Text.pp_word pl in
-        let lstart, lend = Srcloc.as_lexloc w'.loc in
-        (Grammar.STRING t, lstart, lend) :: tokens,
-        prev_right_limit
-
+      | pl -> let t = Grammar.STRING (Format.asprintf "%a" Cobol_preproc.Text.pp_word pl) in
+        emit ~loc:w'.loc t
```

Comment:
```suggestion
      | pl ->
          emit ~loc:w'.loc @@
          Grammar.STRING (Format.asprintf "%a" Cobol_preproc.Text.pp_word pl)
```
Possibly not the right way to handle literals and other remaining tokens. If `pl` is an `alphanum` (as encoded in `cobol_preproc`, possibly that's just the type you should associate with token `Grammar.STRING` (instead of a `string`). So you wouldn't need `asprintf` and such.

In the other cases (`CDirWord`, etc), it's better to list in the match and return a diagnostic about unexpected tokens (to be accumulated along with the list of SQL tokens).